### PR TITLE
Reattach kernel driver in hidapi-libusb

### LINKED ISSUE #110: hidapi-libusb don't reattach kernel driver
For instance in case of following code, the hid device (mouse) freeze after the program exits. Please change the VendorID and ProductID in the code during testing.

```
#ifdef WIN32
#include <windows.h>
#endif
#include <stdio.h>
#include <stdlib.h>
#include <hidapi/hidapi.h>

#define MAX_STR 255

int main(int argc, char* argv[])
{
	int res;
	unsigned char buf[65];
	wchar_t wstr[MAX_STR];
	hid_device *handle;
	int i;

	// Initialize the hidapi library
	res = hid_init();

	if (res==0)
		wprintf(L"HID Initialized.\n");

	// Open the device using the VID, PID,
	// and optionally the Serial number.
	handle = hid_open(0x46d, 0xc534, NULL);
	if (handle != NULL)
		wprintf(L"Device Opened with handle %d\n", handle);
	else{
		wprintf(L"Device not Opened.\n", handle);
		exit(1);
	}

	// Read the Manufacturer String
	res = hid_get_manufacturer_string(handle, wstr, MAX_STR);
	wprintf(L"Manufacturer String: %ls\n", wstr);

	// Read the Product String
	res = hid_get_product_string(handle, wstr, MAX_STR);
	wprintf(L"Product String: %ls\n", wstr);

	// Read the Serial Number String
	res = hid_get_serial_number_string(handle, wstr, MAX_STR);
	if(res==0)
		wprintf(L"Serial Number String: (%d) %ls\n", wstr[0], wstr);

	hid_close(handle);

	// Finalize the hidapi library
	res = hid_exit();

	return 0;
}
```
Compiling the program with `-lhdiapi-libusb` result in frozen device after program run (no issue found with `-lhdiapi-hidraw`). Same issue ([link](https://github.com/signal11/hidapi/issues/305)) was filed for signal11/hidapi.

### PR DESCRIPTION
**Pull request by mayanksuman**
Fixes #110

### CODE REVIEW

File: libusb/hid.c [hunk 1]
```diff
@@ -908,6 +913,7 @@ hid_device * HID_API_EXPORT hid_open_path(const char *path)
 #ifdef DETACH_KERNEL_DRIVER
 						/* Detach the kernel driver, but only if the
 						   device is managed by the kernel */
+						dev->is_driver_detached=0;
```
Comment (Qbicz) [inline libusb/hid.c]:
```suggestion
						dev->is_driver_detached = 0;
```

File: libusb/hid.c [hunk 2]
```diff
@@ -1247,6 +1257,15 @@ void HID_API_EXPORT hid_close(hid_device *dev)
 	/* release the interface */
 	libusb_release_interface(dev->device_handle, dev->interface);
 
+	/* reacttach the kernel driver if it was detached */
```
Comment (Qbicz) [inline libusb/hid.c]:
```suggestion
	/* reacttach the kernel driver if it was detached */
```
```suggestion
	/* reattach the kernel driver if it was detached */
```

File: libusb/hid.c [hunk 3]
```diff
@@ -1247,6 +1257,15 @@ void HID_API_EXPORT hid_close(hid_device *dev)
...
+#ifdef DETACH_KERNEL_DRIVER
+	if (dev->is_driver_detached){
+		int res = libusb_attach_kernel_driver(dev->device_handle, dev->interface);
+		if (res<0)
+			LOG("Failed to attach the driver to kernel.");
```
Comment (Qbicz) [inline libusb/hid.c]:
```suggestion
			LOG("Failed to reattach the driver to kernel.");
```

File: libusb/hid.c [hunk 4]
```diff
@@ -1247,6 +1257,15 @@ void HID_API_EXPORT hid_close(hid_device *dev)
 	/* release the interface */
 	libusb_release_interface(dev->device_handle, dev->interface);
 
+	/* reacttach the kernel driver if it was detached */
+#ifdef DETACH_KERNEL_DRIVER
+	if (dev->is_driver_detached){
+		int res = libusb_attach_kernel_driver(dev->device_handle, dev->interface);
+		if (res<0)
```
Comment (Qbicz) [inline libusb/hid.c]:
```suggestion
		if (res < 0)
```

Comment (mayanksuman) [reply libusb/hid.c]: All changes have been implemented in the commit 8d4cc17.

**Comment by Youw** [comment]
> Can you squash 2 last commits "fix sth" into previous commits?

I think it is best to squash the entire PR instead - it is a single logical change

**Pull request merged**